### PR TITLE
Upgrade to Lucene-7.4.0-snapshot-2b27dd846a

### PR DESCRIPTION
**Pull request by dnhatn**
This Lucene snapshot supports soft delete and the new merge policy.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/indices/analysis/AnalysisFactoryTestCase.java [hunk 1]
```diff
@@ -228,6 +228,8 @@ private static String toCamelCase(String s) {
         // fake term frequency feature (LUCENE-7854)
         .put("delimitedtermfrequency",    Void.class)
 
+        .put("fixedshingle", Void.class)
```
Comment (jpountz) [inline test/framework/src/main/java/org/elasticsearch/indices/analysis/AnalysisFactoryTestCase.java]: please put a comment with something like `not meant to be exposed as-is, will be used internally to index shingles and speed up phrase queries`

File: server/src/main/java/org/elasticsearch/Version.java [hunk 2]
```diff
@@ -167,7 +167,7 @@
     public static final Version V_6_3_0 = new Version(V_6_3_0_ID, org.apache.lucene.util.Version.LUCENE_7_3_0);
     public static final int V_7_0_0_alpha1_ID = 7000001;
     public static final Version V_7_0_0_alpha1 =
-        new Version(V_7_0_0_alpha1_ID, org.apache.lucene.util.Version.LUCENE_7_3_0);
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/Version.java]: can you also replace the 7_2_1 version constant above to replace it with the one that comes from Lucene?

**Review by jpountz — APPROVED** [review]

**Comment by dnhatn** [comment]
Thanks @jpountz for reviewing.

File: server/src/main/java/org/elasticsearch/Version.java (hunk 2 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/Version.java]: Done - I pushed https://github.com/elastic/elasticsearch/pull/29398/commits/81a33ec3dfd8adbe6a4b54a8f37944d7a7645f52

File: test/framework/src/main/java/org/elasticsearch/indices/analysis/AnalysisFactoryTestCase.java (hunk 1 above)
Comment (dnhatn) [reply test/framework/src/main/java/org/elasticsearch/indices/analysis/AnalysisFactoryTestCase.java]: I added a comment

**Review by jpountz — APPROVED** [review]

**Review by s1monw — APPROVED** [review]

**Comment by jpountz** [comment]
@dnhatn It looks like you forgot to remove the 7.3.0 SHA1 files.

**Comment by dnhatn** [comment]
@jpountz You're correct, I forgot to update the shas when merging. Anyway, I will make another snapshot to include the soft-delete directory reader change.

**Comment by dnhatn** [comment]
Thanks @jpountz and @s1monw.

**Pull request merged**